crates/testing-utils/src/lib.rs

```diff
@@ -1,2 +1,22 @@
+use std::{sync::mpsc, thread, time::Duration};
 pub mod bin;
 pub mod fs;
+
+pub fn panic_after<T, F>(d: Duration, f: F) -> T
+where
+    T: Send + 'static,
+    F: FnOnce() -> T,
+    F: Send + 'static,
+{
+    let (done_tx, done_rx) = mpsc::channel();
+    let handle = thread::spawn(move || {
+        let val = f();
+        done_tx.send(()).expect("Unable to send completion signal");
+        val
+    });
+
+    match done_rx.recv_timeout(d) {
+        Ok(_) => handle.join().expect("test panicked"),
+        Err(_) => panic!("test timeout"),
+    }
+}
```

Comment:
Create a mod named `misc` (`crates/testing-utils/src/misc.rs`) for this function.

Reply:
Thanks for your review,  updated here now~